Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -0,0 +1,74 @@
+#import "SentryNetworkTracker.h"
+#import "SentryOptions+Private.h"
+#import "SentryPerformanceTracker.h"
+#import "SentrySDK+Private.h"
+#import <objc/runtime.h>
+
+static NSString *const SENTRY_NETWORK_REQUEST_TRACKER_SPAN_ID
+    = @"SENTRY_NETWORK_REQUEST_TRACKER_SPAN_ID";
+
+@interface
+SentryNetworkTracker ()
+
+@property (nonatomic, strong) SentryPerformanceTracker *tracker;
+
+@end
+
+@implementation SentryNetworkTracker
+
++ (SentryNetworkTracker *)sharedInstance
+{
+    static SentryNetworkTracker *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.tracker = SentryPerformanceTracker.shared;
+    }
+    return self;
+}
+
+- (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
```

Comment:
`h`: What if someone suspends the task and then calls resume again. Do we handle this edge case properly? Can we please add tests for this.

Reply:
Excellent point. We would have two spans with different status.
What do you think is the best approach? 

PS.: In not setting status in this PR, next step.

Reply:
@marandaneto how do you do that on Java?

Reply:
yeah what @bruno-garcia said, so we dont have to handle such case